LiteCore/Query/SQLiteFleeceFunctions.cc

```diff
@@ -371,20 +371,27 @@ namespace litecore {
             }
         }
 
+        auto comparable = [](valueType vtype1, valueType vtype2) {
+            // pre-condition: vtype2 != kBoolean
+            return   vtype1 == vtype2 ? true
+                   : vtype2 == kNumber ? vtype1 == kBoolean
+                   : false;
```

Comment:
I find this hard to understand, I think it warrants a comment.  Why is it a requirement that vtype2 not be a boolean?  And what is this actually checking?  The ability to compare?

Reply:
vtype2 is targetType that is set according to arg. We don't check the subtype to determine whether it is a boolean or not, because in this context we will treat boolean as int. On the other hand, vtype1 derives from value in the array. As a Value, its type is finer, could be kBoolean. Basically, the vtype2 already converted to int, inside this function, we only need to convert vtype1 if it is kBoolean.

Reply:
The rule is, treat kBoolean as integer before doing the comparison.

Reply:
Basically, the pre-condition means that the 2nd argument has already been converted to integer if it is a kBoolean.

Reply:
That's the kind of thing that warrants a comment in the source.  I don't think I'd figure that out without at least an hour or two of analysis.